File: server/dotnet/sdk-dotnet-example/Business/Services/PaymentLinkService.cs
using Business.DTOs.CreatePaymentLinks;
using Business.Interfaces.SDKClients;
using Business.Interfaces.Services;

namespace Business.Services;

public class PaymentLinkService(IPaymentLinkClient paymentLinkClient) : IPaymentLinkService
{
    public async Task<CreatePaymentLinkResponseDto> CreatePaymentLinkAsync(CreatePaymentLinkRequestDto request)
    {
        CreatePaymentLinkResponseDto responseDto =
            await paymentLinkClient.CreatePaymentLinkAsync(request);
        
        return responseDto;
    }
}

File: server/nodejs/src/business/services/payment-link-service.ts
import { FastifyBaseLogger } from 'fastify';
import { CreatePaymentLinkResponseDto } from '../dtos/payment-link/create-payment-link-response-dto';
import { CreatePaymentLinkRequestDto } from '../dtos/payment-link/create-payment-link-request-dto';
import { IPaymentLinkClient } from '../../infrastructure/interfaces/payment-link-client-interface';
import { IPaymentLinkService } from '../interfaces/payment-link-service-interface';

export const PaymentLinkService: IPaymentLinkService = {
    async createLink(
        client: IPaymentLinkClient,
        requestDto: CreatePaymentLinkRequestDto,
        logger: FastifyBaseLogger,
    ): Promise<CreatePaymentLinkResponseDto> {
        return await client.createPaymentLink(requestDto, logger);
    },
};
